src/test/groovy/hudson/plugins/gradle/injection/BuildScanInjectionMavenIntegrationTest.groovy

```diff
@@ -503,6 +507,84 @@ node {
         getEnvVarFromNodeProperties(slave, MavenBuildScanInjection.JENKINSGRADLEPLUGIN_MAVEN_PLUGIN_CONFIG_EXT_CLASSPATH) == null
     }
 
+    @Unroll
+    @SuppressWarnings("GStringExpressionWithinString")
+    def 'vcs repository pattern injection for pipeline remote project - #pattern #mavenSetup'(String pattern, String mavenSetup) {
+        given:
+        withInjectionConfig {
+            injectionVcsRepositoryPatterns = pattern
+        }
+        createSlaveAndTurnOnInjection()
+        def mavenInstallationName = setupMavenInstallation()
+        def replacedMavenSetup = mavenSetup.replaceAll("mavenInstallationName", mavenInstallationName)
+
+        def pipelineJob = j.createProject(WorkflowJob)
+        pipelineJob.setDefinition(new CpsFlowDefinition("""
+   stage('Build') {
+        node('foo') {
+            $replacedMavenSetup
+                git branch: 'main', url: 'https://github.com/c00ler/simple-maven-project'
+                if (isUnix()) {
+                    sh "env"
+                    sh "mvn package -B"
+                } else {
+                    bat "set"
+                    bat "mvn package -B"
+                }
+            }
+        }
+   }
+""", false))
+
+        when:
+        // In this case, the job had to be executed twice for SCM to be populated (we do the same for Gradle, though
+        // for a different reason). Manual test didn't exhibit this problem, so most probably something caused by a test setup
+        j.buildAndAssertSuccess(pipelineJob)
+        def build = j.buildAndAssertSuccess(pipelineJob)
+
+        then:
+        if (pattern.contains("simple-")) {
```

Comment:
💭 could be replaced by a closure param (see other comment)

Reply:
As discussed, wouldn't work :(